changelog.d/12846.feature

```diff
@@ -0,0 +1 @@
+Experimental: expand `check_event_for_spam` with ability to return additional fields.
```

Comment:
The PR description could use some cross-linking to all of the other PR parts to get context.

And some background and examples on what value this adds. What use case does it unlock in the downstream projects?

Reply:
Is this clearer?

Reply:
I probably should have found a better place to put it :) Moving it into the PR description,

Reply:
@Yoric I feel like all of the PRs just describe the "what" and not the "why". This information is locked away in Backend Internal room at the moment when we should also add it to the PR description here.

The context: https://matrix.to/#/!SGNQGPGUwtcPBUotTL:matrix.org/$8N5pv9nqR0GxqN0xfIt8DSjyJ2Easp6AtorjPfgy3Rs?via=jki.re&via=matrix.org&via=element.io

And the use case why: https://matrix.to/#/!SGNQGPGUwtcPBUotTL:matrix.org/$Yy8aCXkUDIgANpkT5ajXC_WkLHgZ1J3cLmSmWHONKLs?via=jki.re&via=matrix.org&via=element.io

Reply:
So, where is the best way to place these answers? As part of the commit message or of the changelog?

Reply:
PR description for sure but both would be good. When I squash a PR to merge, I usually copy the description in there.

Reply:
@Yoric Friendly ping to add details to the description.

Reply:
To something that has merged 10 days ago? How should I proceed?

Reply:
I've added some words to the description. @Yoric could you check that they match reality (feel free to correct/extend if not)?

Reply:
@Yoric confirmed the accuracy in an internal room https://matrix.to/#/!SGNQGPGUwtcPBUotTL:matrix.org/$HVDGbAX1XZ5sgOHxwV98xibNOTTLJCCVek2Kx7dSfoo?via=jki.re&via=matrix.org&via=element.io

Thanks for updating it @richvdh 🤕